core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -185,33 +196,39 @@ private WeekActivityOverviewDto getWeekActivityOverview(UUID userAnonymizedId, L
 				.collect(Collectors.toSet());
 	}
 
-	private WeekActivityDto createAndSaveWeekInactivity(UserAnonymizedDto userAnonymized, Goal goal, ZonedDateTime startOfWeek,
-			LevelOfDetail levelOfDetail, Set<IntervalInactivityDto> missingInactivities)
+	private WeekActivityDto createAndSaveWeekInactivity(UserAnonymizedDto userAnonymized, LocalDate earliestPossibleDate,
+			Goal goal, ZonedDateTime startOfWeek, LevelOfDetail levelOfDetail, Set<IntervalInactivityDto> missingInactivities)
 	{
-		return WeekActivityDto.createInstanceInactivity(userAnonymized, goal, startOfWeek, levelOfDetail, missingInactivities);
+		return WeekActivityDto.createInstanceInactivity(userAnonymized, earliestPossibleDate, goal, startOfWeek, levelOfDetail,
+				missingInactivities);
 	}
 
-	private void createAndSaveInactivityDays(UserAnonymizedDto userAnonymized, Set<GoalDto> goals, WeekActivityDto weekActivity,
-			Set<IntervalInactivityDto> missingInactivities)
+	private void createAndSaveInactivityDays(UserAnonymizedDto userAnonymized, LocalDate earliestPossibleDate, Set<GoalDto> goals,
+			WeekActivityDto weekActivity, Set<IntervalInactivityDto> missingInactivities)
 	{
-		weekActivity.createRequiredInactivityDays(userAnonymized, goals, LevelOfDetail.WEEK_OVERVIEW, missingInactivities);
+		weekActivity.createRequiredInactivityDays(userAnonymized, earliestPossibleDate, goals, LevelOfDetail.WEEK_OVERVIEW,
+				missingInactivities);
 	}
 
-	private long getTotalPageableItems(Set<BuddyDto> buddies, ChronoUnit timeUnit)
+	private long getTotalPageableItems(UserAnonymizedDto userAnonymized, Set<BuddyDto> buddies, ChronoUnit timeUnit)
 	{
-		return buddies.stream().map(this::getBuddyUserAnonymizedId).map(id -> userAnonymizedService.getUserAnonymized(id))
-				.map(ua -> getTotalPageableItems(ua, timeUnit)).max(Long::compareTo).orElse(0L);
+		LocalDate earliestPossibleDate = determineEarliestBuddyStatusChangeDate(buddies);
+
+		return getTotalPageableItems(userAnonymized, earliestPossibleDate, timeUnit);
 	}
 
-	private long getTotalPageableItems(UserAnonymizedDto userAnonymized, ChronoUnit timeUnit)
+	private LocalDate determineEarliestBuddyStatusChangeDate(Collection<BuddyDto> buddies)
+	{
+		// Buddies collection must be non-empty
+		return buddies.stream().map(BuddyDto::getLastStatusChangeTime).map(LocalDateTime::toLocalDate).min(LocalDate::compareTo)
+				.get();
+	}
+
+	private long getTotalPageableItems(UserAnonymizedDto userAnonymized, LocalDate earliestPossibleDate, ChronoUnit timeUnit)
 	{
 		long activityMemoryDays = yonaProperties.getAnalysisService().getActivityMemory().toDays();
-		Optional<LocalDate> oldestGoalCreationDate = userAnonymized.getOldestGoalCreationTime()
```

Comment:
This property is no longer used by now, is it? Can we remove it and the supporting code around it?

Reply:
Good catch! Removed it.

Reply:
Good catch! Removed it.